File: ConduitLLM.Configuration/Enums/ProviderType.cs
namespace ConduitLLM.Configuration
{
    /// <summary>
    /// Strongly-typed enumeration of supported LLM providers
    /// This is NOT a unique identifer for providers! 
    /// It's really used for determining which provider to use for a given model.
    /// All other references to this enum are likely suspect and should be removed!
    /// </summary>
    public enum ProviderType
    {
        /// <summary>
        /// OpenAI provider (GPT models)
        /// </summary>
        OpenAI = 1,

        /// <summary>
        /// Groq
        /// </summary>
        Groq = 2,

        /// <summary>
        /// Replicate
        /// </summary>
        Replicate = 3,

        /// <summary>
        /// Fireworks AI
        /// </summary>
        Fireworks = 4,

        /// <summary>
        /// OpenAI-compatible generic provider
        /// </summary>
        OpenAICompatible = 5,

        /// <summary>
        /// MiniMax
        /// </summary>
        MiniMax = 6,

        /// <summary>
        /// Ultravox
        /// </summary>
        Ultravox = 7,

        /// <summary>
        /// ElevenLabs (audio)
        /// </summary>
        ElevenLabs = 8,

        /// <summary>
        /// Cerebras (high-performance inference)
        /// </summary>
        Cerebras = 9,

        /// <summary>
        /// SambaNova Cloud (ultra-fast inference provider)
        /// </summary>
        SambaNova = 10,

        /// <summary>
        /// DeepInfra (OpenAI-compatible LLM inference platform)
        /// </summary>
        DeepInfra = 11
    }
}

File: SDKs/Node/Core/src/models/providerType.ts
/**
 * Strongly-typed enumeration of supported LLM providers.
 * These numeric values must match the C# ProviderType enum exactly.
 */
export enum ProviderType {
  /** OpenAI provider (GPT models) */
  OpenAI = 1,
  
  /** Groq */
  Groq = 2,
  
  /** Replicate */
  Replicate = 3,
  
  /** Fireworks AI */
  Fireworks = 4,
  
  /** OpenAI-compatible generic provider */
  OpenAICompatible = 5,
  
  /** MiniMax */
  MiniMax = 6,
  
  /** Ultravox */
  Ultravox = 7,
  
  /** ElevenLabs (audio) */
  ElevenLabs = 8,
  
  /** Cerebras (high-performance inference) */
  Cerebras = 9
}